src/ODataWriter.cs

```diff
@@ -318,7 +319,18 @@ internal string MapValuesToJSon(Dictionary<string, object> row, bool isPatchRequ
     public static string GetTheDateTimeInZeroTimeZone(object dateTimeObject, bool isEdmDate)
     {
         var dateTime = Converter.ToDateTime(dateTimeObject);
-        DateTime dateTimeInUtc = TimeZoneInfo.ConvertTimeToUtc(dateTime);
+        DateTime dateTimeInUtc;
+        if (SqlDateTime.MinValue.Value == dateTime || DateTime.MinValue == dateTime)
+        {
+            return DateTime.MinValue.ToString("yyyy-MM-dd");
+        }
+        else if (SqlDateTime.MaxValue.Value == dateTime || DateTime.MaxValue == dateTime)
+        {
+            return DateTime.MaxValue.ToString("yyyy-MM-dd");
+        }
+
+        dateTimeInUtc = TimeZoneInfo.ConvertTimeToUtc(dateTime);
+
         if (dateTimeInUtc.TimeOfDay.TotalMilliseconds > 0 && !isEdmDate)
         {
             return dateTimeInUtc.ToString("yyyy-MM-ddTHH:mm:ss.fff", CultureInfo.InvariantCulture) + "z";
```

Comment:
We don't always skip "z", is that intentional?
We only seem to skip it when handling min or max dates

Reply:
yeah only when we reach max or min datetimes. as they need to look like "0001-01-01" or "9999-12-31" without "z", other dates needs the "z" for zero-datetime right? :)